src/main/java/com/theelixrlabs/healthcare/service/PatchPatientService.java

```diff
@@ -0,0 +1,80 @@
+package com.theelixrlabs.healthcare.service;
+
+import com.theelixrlabs.healthcare.constants.MessageConstants;
+import com.theelixrlabs.healthcare.constants.PatientConstants;
+import com.theelixrlabs.healthcare.dto.PatientDto;
+import com.theelixrlabs.healthcare.exceptionHandler.CustomException;
+import com.theelixrlabs.healthcare.exceptionHandler.ResourceNotFoundException;
+import com.theelixrlabs.healthcare.model.PatientModel;
+import com.theelixrlabs.healthcare.repository.DoctorRepository;
+import com.theelixrlabs.healthcare.repository.PatientRepository;
+import com.theelixrlabs.healthcare.utility.MessageUtil;
+import com.theelixrlabs.healthcare.validation.Validator;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Service;
+import java.util.Optional;
+import java.util.UUID;
+
+@Service
+public class PatchPatientService {
+    private final PatientRepository patientRepository;
+    private final Validator validator;
+    private final MessageUtil messageUtil;
+    private final DoctorRepository doctorRepository;
+
+    public PatchPatientService(PatientRepository patientRepository, Validator validator, MessageUtil messageUtil, DoctorRepository doctorRepository) {
+        this.patientRepository = patientRepository;
+        this.validator = validator;
+        this.messageUtil = messageUtil;
+        this.doctorRepository = doctorRepository;
+    }
+
+    /**
+     * Replace existing patient details based on request.
+     *
+     * @param patientId  Patient unique UUID
+     * @param patientDto Data transfer object containing patient information.
+     * @return The replaced patient details.
+     * @throws CustomException If validation fails or patient already exists.
+     */
+    public PatientDto patchPatientById(String patientId, PatientDto patientDto) throws CustomException {
```

Comment:
Instead of throwing CustomException can we throw PatientException?

Reply:
Created PatientException ,DoctorException and DoctorPatientAssignmentException to handle specific exception